geometry.scad

```diff
@@ -1606,11 +1734,29 @@ function centroid(poly) =
 //   eps = Acceptable variance.  Default: `EPSILON` (1e-9)
 function point_in_polygon(point, path, eps=EPSILON) =
     // Original algorithm from http://geomalgorithms.com/a03-_inclusion.html
+    assert( is_vector(point,2) && is_path(path,dim=2) && len(path)>2,
+            "The point and polygon should be in 2D. The polygon should have more that 2 points." )
+    assert( is_finite(eps) && eps>=0, "Invalid tolerance." )
     // Does the point lie on any edges?  If so return 0.
-    sum([for(i=[0:1:len(path)-1]) let(seg=select(path,i,i+1)) if(!approx(seg[0],seg[1],eps=eps)) point_on_segment2d(point, seg, eps=eps)?1:0]) > 0? 0 :
+    let(
+        on_brd = [for(i=[0:1:len(path)-1]) 
+                    let( seg = select(path,i,i+1) ) 
+                    if( !approx(seg[0],seg[1],eps=eps) ) 
+                        point_on_segment2d(point, seg, eps=eps)? 1:0 ]
+        )
+    sum(on_brd) > 0? 0 :
     // Otherwise compute winding number and return 1 for interior, -1 for exterior
```

Comment:
Do we need to have an option to specify whether we want to use Even-Odd or Non-Zero winding rules?

Reply:
I am not sure I understand the options.

Reply:
Even-Odd Winding Rule for insideness: https://en.wikipedia.org/wiki/Even–odd_rule
Nonzero Winding Rule: https://en.wikipedia.org/wiki/Nonzero-rule

Reply:
![440px-Even-odd_and_non-zero_winding_fill_rules](https://user-images.githubusercontent.com/350001/90716045-cc447e00-e260-11ea-9e39-b8914f354010.png)
Even-Odd on the bottom left.  Nonzero on the bottom right.

Reply:
Aha, I knew the rules, not the names. It seems that OpenSCAD uses the Non-zero rule in 2D. I will think how to implement both.

Reply:
I have an implementation but to insert it now would require changes in calls from other files. It will be left for a second round.

Reply:
I'm fine with deferring this.